Paintroid/src/main/java/org/catrobat/paintroid/iotasks/LoadImageAsync.kt

```diff
@@ -27,74 +27,93 @@ import android.webkit.MimeTypeMap
-				resolver.getType(uri)
-			} else {
-				val fileExtension = MimeTypeMap.getFileExtensionFromUrl(uri.toString())
-				MimeTypeMap.getSingleton().getMimeTypeFromExtension(fileExtension.toLowerCase(Locale.US))
-			}
-			val returnValue: BitmapReturnValue = if (mimeType == "application/zip" || mimeType == "application/octet-stream") {
-				OpenRasterFileFormatConversion.importOraFile(resolver, uri, context.get())
-			} else {
-				if (scaleImage) {
-					FileIO.getScaledBitmapFromUri(resolver, uri, context.get())
-				} else {
-					FileIO.getBitmapReturnValueFromUri(resolver, uri, context.get())
-				}
-			}
-			returnValue
-		} catch (e: IOException) {
-			Log.e(TAG, "Can't load image file", e)
-			null
-		} catch (e: NullPointerException) {
-			Log.e(TAG, "Can't load image file", e)
-			null
-		}
-	}
+    private fun getMimeType(uri: Uri, resolver: ContentResolver): String? =
+        if (uri.scheme == ContentResolver.SCHEME_CONTENT) {
+            resolver.getType(uri)
+        } else {
+            val fileExtension = MimeTypeMap.getFileExtensionFromUrl(uri.toString())
+            MimeTypeMap.getSingleton()
+                .getMimeTypeFromExtension(fileExtension.toLowerCase(US))
+        }
 
-	override fun onPostExecute(result: BitmapReturnValue?) {
-		val callback = callbackRef.get()
-		if (callback != null && !callback.isFinishing) {
-			callback.onLoadImagePostExecute(requestCode, uri, result)
-		}
-	}
+    private fun getBitmapReturnValue(uri: Uri, resolver: ContentResolver): BitmapReturnValue {
+        val mimeType: String? = getMimeType(uri, resolver)
+        return if (mimeType == "application/zip" || mimeType == "application/octet-stream") {
+            OpenRasterFileFormatConversion.importOraFile(
+                resolver,
+                uri,
+                context.get()
+            )
+        } else {
+            if (scaleImage) {
+                FileIO.getScaledBitmapFromUri(resolver, uri, context.get())
+            } else {
+                FileIO.getBitmapReturnValueFromUri(resolver, uri, context.get())
+            }
+        }
+    }
 
-	interface LoadImageCallback {
-		fun onLoadImagePostExecute(requestCode: Int, uri: Uri?, result: BitmapReturnValue?)
-		fun onLoadImagePreExecute(requestCode: Int)
-		val contentResolver: ContentResolver
-		val isFinishing: Boolean
-	}
+    @Suppress("Detekt.TooGenericExceptionCaught")
+    override fun doInBackground(vararg params: Void?): BitmapReturnValue? {
+        val callback = callbackRef.get()
+        if (callback == null || callback.isFinishing) {
+            return null
+        }
+        if (uri == null) {
+            Log.e(TAG, "Can't load image file, uri is null")
+            return null
+        }
+        return try {
+            val resolver = callback.contentResolver
+            FileIO.filename = "image"
+            val returnValue: BitmapReturnValue = getBitmapReturnValue(uri, resolver)
+            returnValue
```

Comment:
to save some codelines `returnValue` could be rewritten to

`getBitmapReturnValue(uri, callback.resolver)`